### PR TITLE
break snake.ino

### PR DESCRIPTION
**Pull request by adrlau**
a feature update towards snake game, with braking bugs.

### CODE REVIEW

**Review by Joakim-droid — APPROVED** [review]

File: spill/snake.ino [hunk 1]
```diff
@@ -9,10 +9,14 @@ const int YPin = 34;
 
 const int w = tft.width(); // 320
 const int h = tft.height(); // 480
-char currentDirection;
+char currentDirection = 'R';
 int xPos = 0;
 int yPos = 0;
 
+int positions[1][2] = [[0,0]]; // a list with all cordinates.
```
Comment (D4ndellion) [inline spill/snake.ino:16]: positions is a one element list containing a two element list containing integers. Initialized to what?

Why is it only one big? What was the point of this?

File: spill/snake.ino [hunk 2]
```diff
@@ -77,8 +79,36 @@ void drawGrid() {
...
   int oldX = xPos;
   int oldY = yPos;
+
+
+if (growing)
+{
+  int newPositions[size+1][2];
+  //opying last element to new last
+  newPositions[size+1][0] = positions[size][0];
+  newPositions[size+1][1] = positions[size][1];
```
Comment (D4ndellion) [inline spill/snake.ino:92]: size is a buffer overflow, `size-1`

File: spill/snake.ino [hunk 3]
```diff
@@ -77,8 +79,36 @@ void drawGrid() {
...
+  for (int i = size-1; i > 0; i--)
+  {
+      newPositions[i+1][0] = positions[i][0];
+      newPositions[i+1][1] = positions[i][1];
+  }
```
Comment (D4ndellion) [inline spill/snake.ino:97]: Moving the elements to the new list can be a memcpy with an offset of siezof(int[2])

File: spill/snake.ino [hunk 4]
```diff
@@ -77,8 +79,36 @@ void drawGrid() {
...
+  int positions = newPositions;
```
Comment (D4ndellion) [inline spill/snake.ino:98]: setting an int to an array will give you just the first in in the first element of the list.

`positions = newpositions` if that doesn't work im not quite sure what to do. Will have to try

File: spill/snake.ino [hunk 5]
```diff
@@ -77,8 +79,36 @@ void drawGrid() {
...
+  size ++;
+}else{
+  //if not growing in size overvriting last digit
+  for (int i = size-1; i > 0; i--)
+  {
+      positions[i+1][0] = positions[i][0];
+      positions[i+1][1] = positions[i][1];
```
Comment (D4ndellion) [inline spill/snake.ino:105]: Buffer overflow

**Pull request merged**